### PR TITLE
chore(design-system/Card): Refactor Card component

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
Card component implementation is dated and it uses styled components.

**What is the chosen solution to this problem?**
Rework it

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/design-system/src/components/Card/Card.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+				flexDirection: 'column',
+				alignItems: 'center',
+				maxWidth: '50rem',
+				padding: '2rem',
+			}}
+		>
+			{children}
+		</div>
+	);
+}
```
Comment (Benrajalu) [inline packages/design-system/src/components/Card/Card.stories.tsx:23]:
```suggestion
```

File: packages/design-system/src/components/Card/Card.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+### Sizing
+
+WIP
+
+### Variations
+
+**Default card**
+
+<Canvas>
+	<Story height="5rem" story={Stories.DefaultCard} />
```
Comment (Benrajalu) [inline packages/design-system/src/components/Card/Card.stories.mdx:34]:
```suggestion
	<Story story={Stories.DefaultCard} />
```

File: packages/design-system/src/components/Card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import { StackVertical } from '../Stack';
+
+import theme from './Card.scss';
+
+interface CardPropsType {
+	header: React.ReactNode;
```
Comment (Benrajalu) [inline packages/design-system/src/components/Card/Card.tsx:8]:
```suggestion
	header?: React.ReactNode;
```

File: packages/design-system/src/components/Card/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	children: React.ReactNode;
+}
+
+function Card({ header, children }: CardPropsType) {
+	return (
+		<div className={theme.card}>
+			<StackVertical gap="L" alignContent="center">
+				<div className={theme.card__header}>{header}</div>
```
Comment (Benrajalu) [inline packages/design-system/src/components/Card/Card.tsx:16]:
```suggestion
				{ header && <div className={theme.card__header}>{header}</div> }
```

File: packages/design-system/src/components/Card/Card.scss [hunk 5]
```diff
@@ -0,0 +1,15 @@
+@use '~@talend/design-tokens/lib/tokens';
+
+.card {
+	width: 100%;
+	height: 100%;
```
[the remark this answers is not in the document]
Comment (sgangnant) [reply packages/design-system/src/components/Card/Card.scss:5]: Not with a this flex parent

**Comment by Benrajalu** [comment]
Please wait for a @mhuet visual review as well

**Review by mhuet — COMMENTED** [review]
Visually approved by Monia ✅

File: packages/design-system/src/templates/Card.scss [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+	:global(.layout__main) {
+		background: none;
+		padding: 5rem;
+	}
+
+	&__card {
+		width: 100vw;
+		margin: 0 auto;
+		max-width: 65rem;
```
Comment (jsomsanith-tlnd) [inline packages/design-system/src/templates/Card.scss:17]: width: 100vw + max-width: 65 rem --> width: 65rem
```suggestion
		width: 65rem;
```

File: packages/design-system/src/components/Card/Card.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+interface CardPropsType {
+	header?: React.ReactNode;
+	children: React.ReactNode;
+}
+
+function Card({ header, children }: CardPropsType) {
+	return (
+		<div className={theme.card}>
+			<StackVertical gap="L" alignContent="center">
+				{header && <div className={theme.card__header}>{header}</div>}
```
[the remark this answers is not in the document]
Comment (sgangnant) [reply packages/design-system/src/components/Card/Card.tsx:16]: I'm afraid header's tag scope is too wide in comparison to what a card can be used for (wrap a page's main content vs showing simple infromation - in a dashboard for example).
But if it's semantically acceptable, I can change it no objection

[the remark this answers is not in the document]
Comment (jsomsanith-tlnd) [reply packages/design-system/src/components/Card/Card.tsx:16]: The `<header>` element can be used to specify the heading of its related content in a section, not necessarily the page.
[header link](https://www.geeksforgeeks.org/html5-header-tag/)

[the remark this answers is not in the document]
Comment (sgangnant) [reply packages/design-system/src/components/Card/Card.tsx]: Updated, thanks!

**Review by jsomsanith-tlnd — APPROVED** [review]

**Pull request merged**